### DISCUSSION TITLE
LINQ filtering on dates

### DISCUSSION THREAD

**Discussion by wonderplayer** [pnp/pnpcore#392] [Q&A] [answered]
Hello,
I would like to know if it is possible to compare item dates in LINQ query?
I can compare simpler values, like string:
```
foreach (var item in ctx.Web.Lists.GetById(new Guid(notification.Resource)).Items.QueryProperties(i => i.Title, i => i["Modified"], i => i["Created"]).Where(i => (string)i.Values["Title"] == "6999"))
{
  log.LogInformation($"LINQ --> '{item.Values.Count}'");
}
```

But if I try to filter on dates, I get exception: `PnP.Core: SharePoint Rest service exception.`:
`var items1 = ctx.Web.Lists.GetById(new Guid(notification.Resource)).Items.QueryProperties(i => i.Title, i => i["Modified"], i => i["Created"]).Where(i => (DateTime)i.Values["Created"] > date);`

So, it's not supported now, or I'm doing something incorrectly?

Thank you!

**Comment by jansenbe** [comment, marked as answer]
@wonderplayer : we've built a custom LINQ provider so we could translate the LINQ expression into a SP/Graph query. The current implementation indeed does only support basic expressions, so what you see is expected. Depending on your scenario doing a CAML query in combination with the ListDataAsStream method might be an alternative solution for you: https://pnp.github.io/pnpcore/using-the-sdk/listitems-intro.html#using-paging-with-listdataasstream
